confmap/provider/envprovider/provider.go

```diff
@@ -9,28 +9,41 @@ import (
 	"os"
 	"strings"
 
+	"go.uber.org/zap"
+
 	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/confmap/provider/internal"
 )
 
 const schemeName = "env"
 
-type provider struct{}
+type provider struct {
+	logger *zap.Logger
+}
 
 // NewWithSettings returns a new confmap.Provider that reads the configuration from the given environment variable.
 //
 // This Provider supports "env" scheme, and can be called with a selector:
 // `env:NAME_OF_ENVIRONMENT_VARIABLE`
-func NewWithSettings(confmap.ProviderSettings) confmap.Provider {
-	return &provider{}
+func NewWithSettings(a confmap.ProviderSettings) confmap.Provider {
+	return &provider{
+		logger: a.Logger,
+	}
```

Comment:
```suggestion
func NewWithSettings(ps confmap.ProviderSettings) confmap.Provider {
	return &provider{
		logger: ps.Logger,
	}
```
Nit: could we make the variable name an abbreviation of the type? It's not immediately clear to me when reading this what `a` stands for. Really minor for how short this function is, but if it were to grow in the future it could be a bit of mental overhead.

Reply:
great point!  fixed